infra/worker/requestWorker.go

```diff
@@ -0,0 +1,207 @@
+		saveAsInvalidHash(req, config, "empty event", models.EmptyEvent)
+		return
+	}
+	block, blockErr := evmClient.BlockByNumber(helpers.Uint64ToBigInt(parsedLogs[0].Raw.BlockNumber))
+	if blockErr != nil {
+		logEntry.WithError(blockErr).Error("fetch block fail")
+		return
+	}
+	crossInfoArr, crossItemsArr := buildCrossInfo(block, parsedLogs, req)
+	allItemCount := 0
+	allTxErr := database.DB.Transaction(func(tx *gorm.DB) error {
+		infoE := tx.Create(crossInfoArr).Error
+		var allItems []models.CrossItem
+		if infoE != nil {
+			return infoE
+		}
+		for i, info := range crossInfoArr {
+			for _, item := range crossItemsArr[i] {
+				item.CrossInfoId = info.Id
+				allItems = append(allItems, item)
+			}
+			allItemCount += len(crossItemsArr[i])
+		}
+		itemE := tx.Create(allItems).Error
+		if itemE != nil {
+			return itemE
+		}
+		configE := tx.Model(&config).Updates(&models.Config{
+			Name:    models.CrossReqId,
+			Content: fmt.Sprint(req.Id),
+		}).Error
+		return configE
+	})
+	if allTxErr != nil {
+		logEntry.WithError(allTxErr).Error("save crossing info error")
+		time.Sleep(time.Second)
+		return
+	}
+	logEntry.WithFields(logrus.Fields{
+		"count": len(crossInfoArr), "itemCount": allItemCount,
+	}).Infof("crossing info saved successfully")
+}
+
+func saveAsInvalidHash(req models.CrossRequest, config models.Config, result string, status int) {
+	e := database.DB.Transaction(func(tx *gorm.DB) error {
+		err1 := tx.Model(&req).Updates(&models.CrossRequest{
+			Id:     req.Id,
+			Status: status,
+			Result: result,
+		}).Error
+		if err1 != nil {
+			return err1
+		}
+		err2 := tx.Model(&config).Updates(&models.Config{
+			Name:    models.CrossReqId,
+			Content: fmt.Sprint(req.Id),
+		}).Error
+		return err2
+	})
+	if e != nil {
+		logrus.Errorf("update req status fail:", e)
+	}
+}
+
+func buildCrossInfo(block *types.Block, parseLogs []*vault.VaultCrossRequest, req models.CrossRequest) ([]models.CrossInfo, [][]models.CrossItem) {
+	blockTime := time.Unix(int64(block.Time()), 0)
+	var crossInfoArr []models.CrossInfo
+	var crossItemsArr = make([][]models.CrossItem, len(parseLogs))
+
+	for infoIndex, crossLog := range parseLogs {
+		var crossInfo = models.CrossInfo{
+			SourceChain:    req.ChainId,
+			TxnHash:        req.Hash,
+			Asset:          crossLog.Asset.Hex(),
+			From:           crossLog.From.Hex(),
+			TargetChain:    crossLog.ToChainId.Int64(),
+			TargetContract: crossLog.TargetContract.Hex(),
+			UserNonce:      crossLog.UserNonce.Int64(),
+			BlockNumber:    crossLog.Raw.BlockNumber,
+			BlockTime:      &blockTime,
+		}
+		crossInfoArr = append(crossInfoArr, crossInfo)
+		var itemArr = make([]models.CrossItem, len(crossLog.TokenIds))
+		for i, tokenId := range crossLog.TokenIds {
+			var item = models.CrossItem{
+				TokenId: common.BigToHash(tokenId).Hex(),
+				Amount:  common.BigToHash(crossLog.Amounts[i]).Hex(),
+				Uri:     crossLog.Uris[i],
+			}
+			itemArr[i] = item
+			logrus.Infof("token is %v\n", tokenId)
```

Comment:
this log info is too vague. besides, I think use `debug` log level would be fine.